l3kernel/testfiles/m3sort002.lvt

```diff
@@ -50,7 +60,91 @@
 \TIMO
 
 \TESTEXP { More~expandable~sorting }
-  { \exp_args:No \tl_sort:nN \l_tmpa_tl \test_compare:nnTF }
+  {
+    \exp_args:No \tl_sort:nN \l_tmpa_tl \test_compare:nnTF
+  }
+
+\OMIT
+\prg_set_conditional:Npnn \test_compare:nn #1#2 { TF }
+  {
+    \if_int_compare:w 
+        \__str_if_eq:nn { \tl_to_str:n {#1} } { \tl_to_str:n {#2} }
+        > 0 \exp_stop_f:
+      \prg_return_false:
+    \else:
+      \prg_return_true:
+    \fi:
+  }
+\TIMO
+
+\TESTEXP { Sorting~unbalanced~conditionals }
+  {
+    \tl_sort:nN { \if_true: \if_false: \else: \fi: } \test_compare:nnTF
+  }
+
+\OMIT
+\box_new:N \l_test_box
+\TIMO
 
+\TEST { halign~robust }
+  {
+    \vbox_set:Nn \l_test_box
+      {
+        \tex_halign:D { \TYPE{0} # \TYPE{1} & # \ERROR \tex_cr:D
+          \exp_after:wN \use_none:nnnn
+          \exp:w \exp_end_continue_f:w
+          \tl_sort:nN { & & \tex_cr:D \tex_cr:D } \test_compare:nnTF
+          \tex_cr:D }
+      }
+  }
+
+\OMIT
+\prg_set_conditional:Npnn \test_compare:nn #1#2 { TF }
+  {
+    \int_compare:nNnTF {#1} > {#2}
+      { \prg_return_false: } { \prg_return_true: }
+  }
+\clist_new:N \l_test_many_clist
+\clist_new:N \l_test_I_clist
+\clist_new:N \l_test_II_clist
+\clist_new:N \l_test_III_clist
+\clist_new:N \l_test_IV_clist
+\clist_set:Nn \l_test_I_clist { \ZERO }
+\clist_set:Nn \l_test_II_clist { 1,\ZERO }
+\clist_set:Nn \l_test_III_clist { 1,\ZERO,2 }
+\clist_set:Nn \l_test_IV_clist { 1,\ZERO,3,2 }
+\clist_set:Nn \l_test_many_clist { 5,1,7,8,10,{1~} }
```

Comment:
Could you please add some negative integers, and perhaps more duplicates (e.g., `1,{1~},1,{~1}`) to show whether the sort preserves the order?